osf/models/files.py

```diff
@@ -386,6 +387,10 @@ def delete(self, user=None, parent=None, save=True, deleted_on=None):
         else:
             self.recast(TrashedFile._typedmodels_type)
 
+            Comment = apps.get_model('osf.Comment')
+            if Comment.objects.filter(root_target=self.guids.first()).exists():
```

Comment:
I'm not 100% sure but I think we don't have to check if this exists first. I think it's actually faster if we always perform a single filter, if the query set is empty the update is essentially a noop.

Reply:
I think still calling exist will be faster. Not 100% sure. @sloria any idea?

Reply:
The `exists()` query does look redundant here. The "SELECT" clause will be the same as the `.update()` call, so we'd be doing 4 queries instead of 2.

Reply:
You might want to try running these queries in `inv shell -p` to see what the execution times are.